mypy/newsemanal/semanal_main.py

```diff
@@ -204,6 +217,11 @@ def semantic_analyze_target(target: str,
         return [], analyzer.incomplete
 
 
+def process_patches(graph: 'Graph', scc: List[str]) -> None:
+    for module in scc:
+        graph[module].semantic_analysis_apply_patches()
```

Comment:
I would add a motivational docstring explaining why we do mini-passes for one things, and patches for other. My understanding that there will be quite few tuple types in a typical program, but quite a lot instances, so we a make a traverser for the latter.

Also there is an option to combine these two in a single type fixing traverser pass, and then repeat it until we solved the chicken and egg problem (this would require to somehow mark types as ready). I however didn't think much about this, maybe this will not help.